### PR TITLE
Create Blog “configuring-azure-ad-with-long-term-token-for-scim-provisiong”

### PR DESCRIPTION
**Pull request by meenakris**
Automatically generated by Netlify CMS

### CODE REVIEW

**Comment by prabhum456** [comment]
1) Should we capture the attribute changes on the Azure application? , It will help since it's for GLC customers and the attributes are related to the SCIM server.
2) Screen shot in the blog is not clear for me, bcoz the PR its clear :) .

**Review by choukden — CHANGES_REQUESTED** [review]
Hi Meena,

Thank you very much for your great contribution to the HPE Developer Community.
Here, you will find some "technical" suggestions for your consideration and to incorporate into the blog post.

Dale Rensing, our editorial consultant, will go through the editorial review process in the coming days or so. Please expect additional suggestions for changes from the editorial/grammatical standpoints.

To answer your question "Should we capture the attribute changes on the Azure application?", I would say YES. 

And for the picture, I am not sure why the image is not rendered in the preview. Please go through my comments and see if that could help fixing the issue becauseI wonder if the issue could not be caused by a couple of backslash in your post.
Or, simply remove the picture, switch the CMS editor to "**Rich text**" and add the picture again as per the guide below (See **tip #5**):

https://github.com/hpe-dev-incubator/hpe-dev-portal/blob/master/docs/ContributorGuide-v2.md#tips-and-trick-using-the-cms-editor
 
Here are **some tips** to assist you with the review of the suggestions and how to acknowledge them in GitHub:

- Navigate to **File Changed** area to visualize the comments from start to the end (You may see “Load Diff” for large blog post entry, in this case just click “Load Diff”), or navigate to **Conversation** area to visualize the comments per reviewer.

- As you incorporate the suggestions for edit in your blog post, please click on button Resolve Conversation in GitHub for each suggestion you incorporated.

Note we recommend you to continue doing edits of the blog post entry in the CMS Workflow editor while the entry is in the "**In Review**" column. Moving it back to the "**Drafts**" column will close the Pull Request (PR) in Github and when moving the entry to the “In Review” column again a new PR in Github is created, and we will lose the suggestions for edits.


Thanks!!!

Best regards
Denis

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+title: Configuring Azure AD with Long Term Token for SCIM Provisiong
```
[the remark this answers is not in the document]
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+priority: 9
+author: Meena Krishnamoorthy
+authorimage: /img/Avatar1.svg
+disable: false
+tags:
+  - HPE GreenLake for Private Cloud Enterprise
+---
+Azure Active Directory (Azure AD) is Microsoft's cloud-based identity and access management service, designed to simplify user authentication and authorization across various applications and platforms. It offers a centralized solution for managing user identities, enforcing security policies, and facilitating seamless access to cloud-based resources. Azure AD automatic user provisioning simplifies the creation, maintenance, and removal of user identities in SaaS applications based on business rules.
+
+The Azure AD provisioning service provisions users to GreenLake portal by connecting to the user management API endpoints provided by GreenLake IAM. These user management API endpoints allow Azure AD to programmatically create, update, and remove users and groups. The Azure AD provisioning service uses GreenLake tenant API token to provision users and groups to GreenLake IAM.  The HPE tenant API tokens are only valid for fifteen minutes. Because Azure AD cannot automatically renew the token, long-term tokens are required.\
```
[the remark this answers is not in the document]
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

[the remark this answers is not in the document]
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+\
+I'll explain the process for configuring Azure AD to use a long-term token for user and group provisioning.
+
+## S﻿teps to configure long-term token in Azure AD
+
+1. A﻿pply SCIM proxy token contributor role to IAM user/group
```
[the remark this answers is not in the document]
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

**Comment by meenakris** [comment]
> 1. Should we capture the attribute changes on the Azure application? , It will help since it's for GLC customers and the attributes are related to the SCIM server.
> 2. Screen shot in the blog is not clear for me, bcoz the PR its clear :) .
>    

Attribute mapping is mainly required for HPCaaS Linux users. I dont think that will be required for PCE customers. 
If need we can update this blog down the line.

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+*   In Azure AD, go to the “Enterprise applications”.
+*   Click the “SSO-Integration” application.
+*   Click the “Provisioning” on the left navigation window.
+*   Click the “Edit provisioning.
+*   Click the “Admin Credentials”.
+*   Update the generated token in the “Secret Token” field.
+*   Update the URL https://sps.us1.greenlake-hpe.com/v1alpha1/scimproxy in the “Tenant URL” field.
+
+![](/img/screenshot-2024-02-14-at-2.08.46 pm.png)
```
[the remark this answers is not in the document]
Comment (choukden) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: The image is now rendered correctly. Thanks!

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+* Save it for use with curl or other REST API client.
+* For example: export BEARER_TOKEN=<paste token value>
+
+**N﻿ote**: This token is valid for 15 minutes after generation
+
+## S﻿tep 3: Create a SCIM proxy token
+
+A SCIM Proxy Token is required for the SCIM integration to work. Run the following curl command to generate the SCIM Proxy token:
+
+curl -H "Authorization: bearer $BEARER_TOKEN" -X POST https://sps.us1.greenlake-hpe.com/v1alpha1/proxytoken
```
[the remark this answers is not in the document]
Comment (choukden) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: The command is not correctly delimited by backticks. You will need to switch to the Markdown editor mode and verify the command is delimited by backticks. 

The preview is here:
https://deploy-preview-2376--hpe-dev-portal.netlify.app/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong/

When correctly delimited by backtick, the rendering should be similar to something like that:

`curl -H "Authorization: bearer $BEARER_TOKEN" -X POST https://sps.us1.greenlake-hpe.com/v1alpha1/proxytoken`

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (choukden) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: The last word in the title has a typo. 
provisiong --> provisioning

**Comment by prabhum456** [comment]
> > 1. Should we capture the attribute changes on the Azure application? , It will help since it's for GLC customers and the attributes are related to the SCIM server.
> > 2. Screen shot in the blog is not clear for me, bcoz the PR its clear :) .
> >    
> 
> Attribute mapping is mainly required for HPCaaS Linux users. I dont think that will be required for PCE customers. If need we can update this blog down the line.

It make sense , if the blogs is for specific to PCE customer .

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+authorimage: /img/Avatar1.svg
+disable: false
+tags:
+  - hpe-greenlake-for-private-cloud-enterprise
+---
+Azure Active Directory (Azure AD) is Microsoft's cloud-based identity and access management service, designed to simplify user authentication and authorization across various applications and platforms. It offers a centralized solution for managing user identities, enforcing security policies, and facilitating seamless access to cloud-based resources. Azure AD automatic user provisioning simplifies the creation, maintenance, and removal of user identities in SaaS applications based on business rules.
+
+The Azure AD provisioning service provisions users to HPE GreenLake portal by connecting to the user management API endpoints provided by HPE GreenLake Identity and Access Management (IAM). These user management API endpoints allow Azure AD to programmatically create, update, and remove users and groups. The Azure AD provisioning service uses HPE GreenLake tenant API token to provision users and groups to HPE GreenLake IAM.  The HPE tenant API tokens are only valid for fifteen minutes. Because Azure AD cannot automatically renew the token, long-term tokens are required.
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:13]: In the first sentence, add the word "the" prior to "...HPE GreenLake portal by..."

In the 3rd sentence, insert the word "an" prior to "...HPE GreenLake tenant API..." and add "the" prior to the end where it says "... HPE GreenLake IAM."

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+I'll explain the process for configuring Azure AD to use a long-term token for user and group provisioning.
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:15]: Suggest adding "In this blog post, " to the beginning here so it says "In this blog post, I'll explain the process..."

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+## S﻿teps to configure long-term token in Azure AD
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:17]: Change "token" to plural "tokens"

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+1. A﻿pply System for Cross-domain Identity Management(SCIM) proxy token contributor role to IAM user/group
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:19]: Need a space between "Management" and "(SCIM)"
Insert the word "the" between "Apply" and "System"

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+2. G﻿et a personal access token
+3. C﻿reate a SCIM proxy token
+4. U﻿pdate the SCIM proxy token and the tenant URL in Azure AD Enterprise Application
+
+## S﻿tep 1: Apply System for Cross-domain Identity Management(SCIM) proxy token contributor role to IAM user/group
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:24]: Add a space between "Management" and "(SCIM)"

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+A﻿ssign "SCIM Proxy Token Contributor" role to the user or user group that will create the long-term token
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:26]: Add a period at the end.

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+* L﻿og in to the HPE GreenLake Cental.
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:28]: Change "in to" to "into".
Check the spelling of "Central"

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+* C﻿lick the "User Management" icon on the top-right corner.
+* S﻿elect user/user group that will generate the SCIM proxy token.
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:30]: Add the word "the" after "Select"

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+* E﻿nable "I confirm that I want to create the assignments listed above".
+* C﻿lick "Create Assignment" button.
+
+**N﻿ote**: This must be applied by an HPE engineer who has HPE GreenLake IAM owner permissions.
+
+## S﻿tep 2: G﻿et a personal access token
+
+An API token issued by the HPE GreenLake Central platform must be used as the Bearer token in the Authorization header of HPE GreenLake Central REST API requests. Perform the following steps to get API access token from HPE GreenLake Central portal:
+
+* Log in to HPE GreenLake Central.
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:43]: Change "in to" to "into"

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+* Click the profile icon on the top-right corner.
+* Select API Access.
+* Copy the API access token.
+* Save it for use with curl or other REST API client.
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:47]: Insert the word "an" before "other"

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+* For example: export BEARER_TOKEN=<paste token value>
+
+**N﻿ote**: This token is valid for 15 minutes after generation
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:50]: Add a period at the end.

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+
+## S﻿tep 3: Create a SCIM proxy token
+
+A SCIM Proxy Token is required for the SCIM integration to work. Run the following curl command to generate the SCIM Proxy token:
+
+`curl -H "Authorization: bearer $BEARER_TOKEN" -X POST https://sps.us1.greenlake-hpe.com/v1alpha1/proxytoken`
+
+**N﻿ote**: This step must be performed once during initial setup and every time a token is deleted.
```
Comment (drensing) [inline content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:58]: Add the word "the" prior to "initial setup"

**Review by drensing — COMMENTED** [review]
Hi there! Thank you very much for your post. Very nice. I hope my comments are helpful.
Could I ask you to please add a summary to the bottom? Each of our blog posts tends to have a section at the end that briefly summarizes what was covered in the post, the benefit that's provided to the reader, and, if possible, a call to action (like "Make sure to come back to read more of my posts on Azure in the HPE Developer blog" or "Refer to this blog post for more information that can help you in similar situations.") And make sure to include links.

**Comment by prabhum456** [comment]
@meenakris  , Please can you add the rotate option for the customer( REST API usage )

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md (hunk 6 above)
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md (hunk 8 above)
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md (hunk 10 above)
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md (hunk 13 above)
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md (hunk 15 above)
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md (hunk 17 above)
Comment (meenakris) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md]: Done

**Comment by meenakris** [comment]
> @meenakris , Please can you add the rotate option for the customer( REST API usage )

Done

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md (hunk 11 above)
Comment (choukden) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:26]: @meenakris 
Could you please add a period at the end of the sentence: Assign SCIM Proxy Token Contributor... long-term token.

File: content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md (hunk 12 above)
Comment (choukden) [reply content/blog/configuring-azure-ad-with-long-term-token-for-scim-provisiong.md:28]: @meenakris 
Could you please check the spelling of HPE GreenLake Central in this sentence: Log in to the HPE GreenLake Central.

**Furthermore,** the **_HPE GreenLake Central_** is now known as **_HPE GreenLake Flex Solutions_**.
You may want to adjust the occurrences of "HPE GreenLake Central" in your blog post to "HPE GreenLake Flex Solutions".

There is a bullet item that mention HPE GreenLake Central in:

**Step 1: apply System for Cross-domain Identity Management (SCIM) proxy token contributor role to IAM user/group**

and

**Step 2: Get a personal access token**

You may want to adjust the text of the first bullet item to something like that:

- Log in to the HPE GreenLake Central, also known as HPE GreenLake Flex Solutions.

If you have an HPE account associated to a HPE GreenLake Workspace, you can log in to the common.cloud.hpe.com, and in the "Services", in the service catalog, you will see information about the "**HPE GreenLake Flex solutions**":

_Formerly known as HPE GreenLake Central, view usage and cost of HPE GreenLake Flex Solutions and manage the Private Cloud Enterprise and High Performance Computing cloud services. For all HPE GreenLake Flex Solutions, this service provides reporting on billing, active capacity planning, and powerful consumption analytics that enable transparency to see costs by service type, department, or other parameters, including public cloud costs. Customers of Private Cloud Enterprise use it to provision and manage virtual, containerized, and bare metal resources, and customers of High Performance Computing use it to operate HPC workloads._

**Review by choukden — CHANGES_REQUESTED** [review]
@meenakris 
Hi Meena,

Thank you very much for taking into account Dale's editorial suggestions for changes.
I have added a couple of comments/suggestions for edit before pushing the blog post to the HPE Developer Community portal.
Could you please have a look at these additional comments.

Once you took into account the suggestions, let me know and I will push the blog post to the HPE Dev Community portal.

Thanks
Denis

**Comment by meenakris** [comment]
> @meenakris , Please can you add the rotate option for the customer( REST API usage )

Done

**Comment by meenakris** [comment]
> @meenakris Hi Meena,
> 
> Thank you very much for taking into account Dale's editorial suggestions for changes. I have added a couple of comments/suggestions for edit before pushing the blog post to the HPE Developer Community portal. Could you please have a look at these additional comments.
> 
> Once you took into account the suggestions, let me know and I will push the blog post to the HPE Dev Community portal.
> 
> Thanks Denis

Thanks Denis, I addressed them.

**Pull request merged**